src/color-swatch/color-swatch.js

```diff
@@ -109,6 +111,26 @@ const Self = class ColorSwatch extends NudeElement {
 			this.style.setProperty("--color", colorString);
 		}
 
+		if (name === "coords") {
+			if (this.color) {
+				this._el.coords?.replaceChildren(); // remove all children
+
+				for (let coord of this.coords) {
+					let [label, properties] = Object.entries(coord)[0];
+					let [space, channel] = properties.split(".");
+					let value;
+
+					if (!channel) {
+						value = this.color[space];
+					}
+					else {
+						value = this.color[space][channel];
+					}
+
+					this._el.coords.insertAdjacentHTML("beforeend", `<div class="coord"><dt>${ label }</dt><dd>${ value }</dd></div>`);
```

Comment:
Let's minimize DOM operations by building this string and then setting the element’s entire `innerHTML` to it. You can build an array with each entry and then set innerHTML to that array with `.join("\n")`

Reply:
More than agreed. 🙂